server/src/session.ts

```diff
@@ -876,33 +861,50 @@ export class Session {
    * Disable the language service if the specified `project` is not Angular or
    * Ivy mode is enabled.
    */
-  private checkProject(project: ts.server.Project) {
+  private disableLanguageServiceIfNecessary(project: ts.server.Project): void {
     const {projectName} = project;
     if (!project.languageServiceEnabled) {
       this.info(
           `Language service is already disabled for ${projectName}. ` +
           `This could be due to non-TS files that exceeded the size limit (${
               ts.server.maxProgramSizeForNonTsFiles} bytes).` +
           `Please check log file for details.`);
-
       return;
     }
 
-    const coreDts = this.checkIsAngularProject(project);
+    const coreDts = this.findAngularCoreDts(project);
     if (coreDts === undefined) {
+      project.disableLanguageService();
+      this.info(
+          `Disabling language service for ${projectName} because it is not an Angular project ` +
+          `('@angular/core' could not be found).`);
+      return;
+    }
+
+    if (isInternalAngularCore(coreDts)) {
       return;
     }
 
     if (this.ivy && isConfiguredProject(project)) {
       // Keep language service disabled until ngcc is completed.
       project.disableLanguageService();
+      const configFilePath = project.getConfigFilePath();
       // Do not wait on this promise otherwise we'll be blocking other requests
-      this.runNgcc(project.getConfigFilePath()).catch((error: Error) => {
-        this.error(error.toString());
-      });
+      this.runNgcc(configFilePath)
+          .then(() => {
+            // Re-enable language service even if ngcc fails, because users could fix
```

Comment:
Does the `.then` execute if ngcc fails?

Reply:
The `runNgcc` function is wrapped in `try-catch-finally` so the promise it returns will never reject.